Make editor sidebar panel expand when clicked

HenintsoaSky opened the pull request:
- Fixes https://github.com/ppy/osu/issues/31064
#### Before:
[Before](https://github.com/user-attachments/assets/a4e38cd6-4118-4dbb-a901-acd873c233d7)

#### After:
[After](https://github.com/user-attachments/assets/846e89c2-98f9-4b46-a05f-f1666b19c9c4)

HenintsoaSky committed:
Make editor sidebar panel expand when clicked

HenintsoaSky committed:
Fix license header missing

HenintsoaSky committed:
Fix CI failures

bdach commented:
On a 15 second check, the following should not be happening:

https://github.com/user-attachments/assets/59130609-a916-4a24-9c8a-6b275aa2bdd9

Ending a drag by releasing mouse expands the sidebar, it shouldn't do that. Maybe the behaviour should be implemented in `OnClick()` instead which I believe is blocked by drag.

Also incidentally once I somehow got into a state wherein the panel would just not collapse back down again. Probably something or other about the timing of the mouse up coinciding with the timing of the delayed expand.

bdach reviewed (changes requested):
as above

HenintsoaSky committed:
Change behaviour

Use OnClick instead of OnMouseUp

HenintsoaSky committed:
Prevent unnecessary collapsing on click

Adjusted OnClick logic in ExpandingToolboxContainer to ensure the container does not collapse when clicked after being expanded.

HenintsoaSky committed:
Change tests

bdach commented on osu.Game.Tests/Visual/Editing/TestSceneExpandingToolboxContainer.cs:
```diff
@@ -0,0 +1,64 @@
+            {
+                InputManager.MoveMouseTo(toolbox.ScreenSpaceDrawQuad.Centre);
+                InputManager.Click(MouseButton.Left);
+                AddAssert("sidebar expands after click", () => toolbox.Expanded.Value);
```
I don't think you want this assert to be nested in a test step.

HenintsoaSky committed:
Address comments

- Removed redundant conditional check:
  `if (!ReceivePositionalInputAt(e.ScreenSpaceMousePosition)) return false;`

- Unnested `AddAssert("sidebar expands after click", () => toolbox.Expanded.Value)`

bdach committed:
Fix broken test & remove weird dependency on config

bdach committed:
Move click-to-expand logic to a higher level

Think it makes more sense there.

bdach commented:
Have essentially rewritten this to get to a state I'd consider passable. Going to wait and see what CI says to my changes before pinging people for a second review.

kongehund commented:
I'm concerned that having `ExpandingContainer` accept the `OnClick` means we can't scroll to the clicked `SettingsToolboxGroup` when clicked, as suggested in my original issue. 

I'm currently looking into having a `SettingsToolboxGroup` accept the click instead, and then notify the `ExpandingContainer` about this to scroll to the clicked group.

kongehund commented:
Not sure how patches work, but see https://github.com/kongehund/osu/tree/refs/heads/expanding-toolbar-click

The changes will scroll the toolbar to the clicked toolbox, also when the toolbar is expanded.

HenintsoaSky commented:
One issue with @kongehund's changes is that when you click near the lowest part of the container, it expands after the click but then collapses again. This happens because the cursor ends up outside the active area, interpreted as "not hovering" over the container. (#31063)

This shows that better: 

https://github.com/user-attachments/assets/f0708212-4a97-48cb-9747-666d385a1293


Given this, I'm unsure if I should merge the changes. RFC
